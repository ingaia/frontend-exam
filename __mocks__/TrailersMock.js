const TrailersMock = [
    {
        "kind": "youtube#playlistItem",
        "etag": "\"OOFf3Zw2jDbxxHsjJ3l8u1U8dz4/o3IEMGHYDQ1g44_lsU3xaCF8Ge0\"",
        "id": "UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41NkI0NEY2RDEwNTU3Q0M2",
        "snippet": {
            "publishedAt": "2014-07-04T21:20:54.000Z",
            "channelId": "UC-BvxTB6agC6mwGzVoos9GQ",
            "title": "The Witcher 3: Wild Hunt - Title Reveal",
            "description": "Der erste offizielle Trailer zu The Witcher 3: Wild Hunt zeigt das bekannte Wolfsmedaillon und gibt einen ersten Vorgeschmack auf die Musik im Spiel.\n\n------------------------------\n\nAlle weiteren Infos zu The Witcher 3: Wild Hunt sowie zur gesamten Witcher-Saga gibt's auf:\nhttp://www.the-witcher.de/",
            "thumbnails": {
                "default": {
                    "url": "https://i.ytimg.com/vi/cCXgQ7-Y9ps/default.jpg",
                    "width": 120,
                    "height": 90
                },
                "medium": {
                    "url": "https://i.ytimg.com/vi/cCXgQ7-Y9ps/mqdefault.jpg",
                    "width": 320,
                    "height": 180
                },
                "high": {
                    "url": "https://i.ytimg.com/vi/cCXgQ7-Y9ps/hqdefault.jpg",
                    "width": 480,
                    "height": 360
                },
                "standard": {
                    "url": "https://i.ytimg.com/vi/cCXgQ7-Y9ps/sddefault.jpg",
                    "width": 640,
                    "height": 480
                },
                "maxres": {
                    "url": "https://i.ytimg.com/vi/cCXgQ7-Y9ps/maxresdefault.jpg",
                    "width": 1280,
                    "height": 720
                }
            },
            "channelTitle": "The-Witcher.de",
            "playlistId": "PL6t93nUFQQ1ZiXMfhPyhjb0PX3LgEVMcF",
            "position": 0,
            "resourceId": {
                "kind": "youtube#video",
                "videoId": "cCXgQ7-Y9ps"
            }
        }
    },
    {
        "kind": "youtube#playlistItem",
        "etag": "\"OOFf3Zw2jDbxxHsjJ3l8u1U8dz4/GwTj3CzqoqxnAC0ELKkzb_a-2Io\"",
        "id": "UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4yODlGNEE0NkRGMEEzMEQy",
        "snippet": {
            "publishedAt": "2014-07-04T21:51:54.000Z",
            "channelId": "UC-BvxTB6agC6mwGzVoos9GQ",
            "title": "The Witcher 3: Wild Hunt - Debut Gameplay-Trailer",
            "description": "In diesem Gameplay-Trailer zeigte man zur E3 2013 erstmals The Witcher 3 in Bewegung und beeindruckte mit toller Grafik.\n\n------------------------------\n\nAlle weiteren Infos zu The Witcher 3: Wild Hunt sowie zur gesamten Witcher-Saga gibt's auf:\nhttp://www.the-witcher.de/",
            "thumbnails": {
                "default": {
                    "url": "https://i.ytimg.com/vi/2T_DG1xnVPg/default.jpg",
                    "width": 120,
                    "height": 90
                },
                "medium": {
                    "url": "https://i.ytimg.com/vi/2T_DG1xnVPg/mqdefault.jpg",
                    "width": 320,
                    "height": 180
                },
                "high": {
                    "url": "https://i.ytimg.com/vi/2T_DG1xnVPg/hqdefault.jpg",
                    "width": 480,
                    "height": 360
                },
                "standard": {
                    "url": "https://i.ytimg.com/vi/2T_DG1xnVPg/sddefault.jpg",
                    "width": 640,
                    "height": 480
                },
                "maxres": {
                    "url": "https://i.ytimg.com/vi/2T_DG1xnVPg/maxresdefault.jpg",
                    "width": 1280,
                    "height": 720
                }
            },
            "channelTitle": "The-Witcher.de",
            "playlistId": "PL6t93nUFQQ1ZiXMfhPyhjb0PX3LgEVMcF",
            "position": 1,
            "resourceId": {
                "kind": "youtube#video",
                "videoId": "2T_DG1xnVPg"
            }
        }
    },
    {
        "kind": "youtube#playlistItem",
        "etag": "\"OOFf3Zw2jDbxxHsjJ3l8u1U8dz4/onAaYJeCvsdOP82FLbPBD0EQZR4\"",
        "id": "UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wOTA3OTZBNzVEMTUzOTMy",
        "snippet": {
            "publishedAt": "2014-07-05T19:37:57.000Z",
            "channelId": "UC-BvxTB6agC6mwGzVoos9GQ",
            "title": "The Witcher 3: Wild Hunt - \"Killing Monsters\"-Cinematic-Trailer",
            "description": "In diesem CGI-Trailer macht Geralt von Riva das, was er am besten kann: Monster schnetzeln.\n\n------------------------------\n\nAlle weiteren Infos zu The Witcher 3: Wild Hunt sowie zur gesamten Witcher-Saga gibt's auf:\nhttp://www.the-witcher.de/",
            "thumbnails": {
                "default": {
                    "url": "https://i.ytimg.com/vi/aIbPBnvGSnc/default.jpg",
                    "width": 120,
                    "height": 90
                },
                "medium": {
                    "url": "https://i.ytimg.com/vi/aIbPBnvGSnc/mqdefault.jpg",
                    "width": 320,
                    "height": 180
                },
                "high": {
                    "url": "https://i.ytimg.com/vi/aIbPBnvGSnc/hqdefault.jpg",
                    "width": 480,
                    "height": 360
                },
                "standard": {
                    "url": "https://i.ytimg.com/vi/aIbPBnvGSnc/sddefault.jpg",
                    "width": 640,
                    "height": 480
                },
                "maxres": {
                    "url": "https://i.ytimg.com/vi/aIbPBnvGSnc/maxresdefault.jpg",
                    "width": 1280,
                    "height": 720
                }
            },
            "channelTitle": "The-Witcher.de",
            "playlistId": "PL6t93nUFQQ1ZiXMfhPyhjb0PX3LgEVMcF",
            "position": 2,
            "resourceId": {
                "kind": "youtube#video",
                "videoId": "aIbPBnvGSnc"
            }
        }
    },
    {
        "kind": "youtube#playlistItem",
        "etag": "\"OOFf3Zw2jDbxxHsjJ3l8u1U8dz4/_nUHMgUyanrie2yiqsfYYaM3f9U\"",
        "id": "UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41MjE1MkI0OTQ2QzJGNzNG",
        "snippet": {
            "publishedAt": "2014-07-05T18:57:01.000Z",
            "channelId": "UC-BvxTB6agC6mwGzVoos9GQ",
            "title": "The Witcher 3: Wild Hunt - VGX-Trailer (2013)",
            "description": "In vorherigen Trailern wurde sie nur angedeutet, in diesem Trailer wird etwas näher auf die dem Spiel namensgebende Wilde Jagd eingegangen.\n\n------------------------------\n\nAlle weiteren Infos zu The Witcher 3: Wild Hunt sowie zur gesamten Witcher-Saga gibt's auf:\nhttp://www.the-witcher.de/",
            "thumbnails": {
                "default": {
                    "url": "https://i.ytimg.com/vi/7T91vrI8hGA/default.jpg",
                    "width": 120,
                    "height": 90
                },
                "medium": {
                    "url": "https://i.ytimg.com/vi/7T91vrI8hGA/mqdefault.jpg",
                    "width": 320,
                    "height": 180
                },
                "high": {
                    "url": "https://i.ytimg.com/vi/7T91vrI8hGA/hqdefault.jpg",
                    "width": 480,
                    "height": 360
                },
                "standard": {
                    "url": "https://i.ytimg.com/vi/7T91vrI8hGA/sddefault.jpg",
                    "width": 640,
                    "height": 480
                },
                "maxres": {
                    "url": "https://i.ytimg.com/vi/7T91vrI8hGA/maxresdefault.jpg",
                    "width": 1280,
                    "height": 720
                }
            },
            "channelTitle": "The-Witcher.de",
            "playlistId": "PL6t93nUFQQ1ZiXMfhPyhjb0PX3LgEVMcF",
            "position": 3,
            "resourceId": {
                "kind": "youtube#video",
                "videoId": "7T91vrI8hGA"
            }
        }
    },
    {
        "kind": "youtube#playlistItem",
        "etag": "\"OOFf3Zw2jDbxxHsjJ3l8u1U8dz4/KANzusO-TxoVIC4vARdovj8YMvY\"",
        "id": "UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wMTcyMDhGQUE4NTIzM0Y5",
        "snippet": {
            "publishedAt": "2014-07-05T16:39:11.000Z",
            "channelId": "UC-BvxTB6agC6mwGzVoos9GQ",
            "title": "The Witcher 3: Wild Hunt - \"The Sword Of Destiny\"",
            "description": "Mit \"The Sword Of Destiny\" präsentierte CD Projekt RED auch zur E3 2014 wieder einen umwerfenden Trailer zum krönenden Abschluss der Geralt-Saga.\n\n------------------------------\n\nTrailer-Analyse (Zwei Teile):\nTeil 1: http://www.the-witcher.de/tw3_trailer_analyse_e314_teil1.html\nTeil 2: http://www.the-witcher.de/tw3_trailer_analyse_e314_teil2.html\n\nDiskussion zum Trailer bei uns im Forum: http://forum.worldofplayers.de/forum/threads/1358814-The-Witcher-3-Wild-Hunt-E3-2014-Trailer-The-Sword-Of-Destiny\n\nAlle weiteren Infos zu The Witcher 3: Wild Hunt sowie zur gesamten Witcher-Saga gibt's auf:\nhttp://www.the-witcher.de/",
            "thumbnails": {
                "default": {
                    "url": "https://i.ytimg.com/vi/1ke3zayKKm8/default.jpg",
                    "width": 120,
                    "height": 90
                },
                "medium": {
                    "url": "https://i.ytimg.com/vi/1ke3zayKKm8/mqdefault.jpg",
                    "width": 320,
                    "height": 180
                },
                "high": {
                    "url": "https://i.ytimg.com/vi/1ke3zayKKm8/hqdefault.jpg",
                    "width": 480,
                    "height": 360
                },
                "standard": {
                    "url": "https://i.ytimg.com/vi/1ke3zayKKm8/sddefault.jpg",
                    "width": 640,
                    "height": 480
                },
                "maxres": {
                    "url": "https://i.ytimg.com/vi/1ke3zayKKm8/maxresdefault.jpg",
                    "width": 1280,
                    "height": 720
                }
            },
            "channelTitle": "The-Witcher.de",
            "playlistId": "PL6t93nUFQQ1ZiXMfhPyhjb0PX3LgEVMcF",
            "position": 4,
            "resourceId": {
                "kind": "youtube#video",
                "videoId": "1ke3zayKKm8"
            }
        }
    }
];

export default TrailersMock;