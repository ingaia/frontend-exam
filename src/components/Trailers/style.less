@import './../../variables.less';

.trailers {
    flex: 5;
    background: linear-gradient(90deg, #000000 0%, rgba(0, 0, 0, 0) 98%);
    border-left: 1px solid @hillary;
    padding: 50px 10px;
    
    .trailers-list {
        display: flex;
        justify-content: center;
        flex-wrap: wrap;
        max-height: 1500px;
        height: 100%;
        overflow-y: scroll;
        list-style-type: none;
        
    }

    .trailers-list::-webkit-scrollbar {
        width: 5px;
    }
    
    .trailers-list::-webkit-scrollbar-track {
        background-color: @hillary;
        background-clip: content-box; 
        border: 2px solid transparent;
        border-radius: 5px;
    }
    
    .trailers-list::-webkit-scrollbar-thumb {
        background-color: @hillary;
        border-radius: 5px;
    }

    .load-more {
        display: flex;
        align-items: center;
        justify-content: center;
        height: 222px;
        width: 375px;
        margin: 50px;
    }
}

@media @mobile {
    .trailers {
        padding: 10px;
        background: none;
        border: none;
        
        .trailers-list {
            flex-wrap: nowrap;
            flex-direction: column;
            align-items: center;
            max-height: unset;        
        }
    }


    .trailers-list::-webkit-scrollbar {
        display: none;
    }
}

@media @tablet {
    .trailers {
        padding: 50px 0px;
    }
}