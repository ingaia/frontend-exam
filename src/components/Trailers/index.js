import React from 'react';

import Trailer from '../Trailer';
import Button from '../Button';
import Loader from '../Loader';

import './style.less';

export default function Trailers(props){
    return (
        <div className="trailers">
            <ul className="trailers-list">
                {props.trailers.map(({id, snippet: {title, thumbnails: { maxres }, resourceId}}) => (
                    <Trailer key={id} title={title} thumbnail={maxres.url} videoId={resourceId.videoId}/>
                ))}
                { props.nextPageToken && !props.loading &&
                    <li className="load-more">
                        <Button className="btn-hillary" label="Load More" onClick={props.loadMore}/>
                    </li>
                }
                { props.loading && 
                    <li className="load-more">
                        <Loader/>
                    </li>
                }
            </ul>
        </div>
    );
}