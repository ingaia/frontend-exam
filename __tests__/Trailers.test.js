import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json'
import Trailers from '../src/components/Trailers';
import TrailersMock from '../__mocks__/TrailersMock';

describe('<Trailers/>', () => {
    let wrapper;
    
    beforeEach(() => {
        wrapper = shallow(<Trailers trailers={TrailersMock}/>);
    });

    it('should render', () => {
        expect(wrapper.hasClass('trailers')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should show the list of trailers with 5 items', () => {
        const list = wrapper.find('.trailers-list');
        
        expect(list.find('Trailer').length).toBe(TrailersMock.length);
    });

    it('should show button load more', () => {
        wrapper.setProps({
            nextPageToken: '1A2B3C',
            loading: false
        });
        
        expect(wrapper.exists('.load-more')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('should show the Loader component', () => {
        wrapper.setProps({
            loading: true
        });
        
        expect(wrapper.exists('Loader')).toBe(true);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});