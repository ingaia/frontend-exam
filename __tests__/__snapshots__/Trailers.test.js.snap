// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Trailers/> should render 1`] = `
<div
  className="trailers"
>
  <ul
    className="trailers-list"
  >
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41NkI0NEY2RDEwNTU3Q0M2"
      thumbnail="https://i.ytimg.com/vi/cCXgQ7-Y9ps/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - Title Reveal"
      videoId="cCXgQ7-Y9ps"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4yODlGNEE0NkRGMEEzMEQy"
      thumbnail="https://i.ytimg.com/vi/2T_DG1xnVPg/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - Debut Gameplay-Trailer"
      videoId="2T_DG1xnVPg"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wOTA3OTZBNzVEMTUzOTMy"
      thumbnail="https://i.ytimg.com/vi/aIbPBnvGSnc/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - \\"Killing Monsters\\"-Cinematic-Trailer"
      videoId="aIbPBnvGSnc"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41MjE1MkI0OTQ2QzJGNzNG"
      thumbnail="https://i.ytimg.com/vi/7T91vrI8hGA/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - VGX-Trailer (2013)"
      videoId="7T91vrI8hGA"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wMTcyMDhGQUE4NTIzM0Y5"
      thumbnail="https://i.ytimg.com/vi/1ke3zayKKm8/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - \\"The Sword Of Destiny\\""
      videoId="1ke3zayKKm8"
    />
  </ul>
</div>
`;

exports[`<Trailers/> should show button load more 1`] = `
<div
  className="trailers"
>
  <ul
    className="trailers-list"
  >
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41NkI0NEY2RDEwNTU3Q0M2"
      thumbnail="https://i.ytimg.com/vi/cCXgQ7-Y9ps/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - Title Reveal"
      videoId="cCXgQ7-Y9ps"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4yODlGNEE0NkRGMEEzMEQy"
      thumbnail="https://i.ytimg.com/vi/2T_DG1xnVPg/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - Debut Gameplay-Trailer"
      videoId="2T_DG1xnVPg"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wOTA3OTZBNzVEMTUzOTMy"
      thumbnail="https://i.ytimg.com/vi/aIbPBnvGSnc/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - \\"Killing Monsters\\"-Cinematic-Trailer"
      videoId="aIbPBnvGSnc"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41MjE1MkI0OTQ2QzJGNzNG"
      thumbnail="https://i.ytimg.com/vi/7T91vrI8hGA/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - VGX-Trailer (2013)"
      videoId="7T91vrI8hGA"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wMTcyMDhGQUE4NTIzM0Y5"
      thumbnail="https://i.ytimg.com/vi/1ke3zayKKm8/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - \\"The Sword Of Destiny\\""
      videoId="1ke3zayKKm8"
    />
    <li
      className="load-more"
    >
      <Button
        className="btn-hillary"
        label="Load More"
      />
    </li>
  </ul>
</div>
`;

exports[`<Trailers/> should show the Loader component 1`] = `
<div
  className="trailers"
>
  <ul
    className="trailers-list"
  >
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41NkI0NEY2RDEwNTU3Q0M2"
      thumbnail="https://i.ytimg.com/vi/cCXgQ7-Y9ps/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - Title Reveal"
      videoId="cCXgQ7-Y9ps"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4yODlGNEE0NkRGMEEzMEQy"
      thumbnail="https://i.ytimg.com/vi/2T_DG1xnVPg/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - Debut Gameplay-Trailer"
      videoId="2T_DG1xnVPg"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wOTA3OTZBNzVEMTUzOTMy"
      thumbnail="https://i.ytimg.com/vi/aIbPBnvGSnc/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - \\"Killing Monsters\\"-Cinematic-Trailer"
      videoId="aIbPBnvGSnc"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi41MjE1MkI0OTQ2QzJGNzNG"
      thumbnail="https://i.ytimg.com/vi/7T91vrI8hGA/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - VGX-Trailer (2013)"
      videoId="7T91vrI8hGA"
    />
    <Trailer
      key="UEw2dDkzblVGUVExWmlYTWZoUHloamIwUFgzTGdFVk1jRi4wMTcyMDhGQUE4NTIzM0Y5"
      thumbnail="https://i.ytimg.com/vi/1ke3zayKKm8/maxresdefault.jpg"
      title="The Witcher 3: Wild Hunt - \\"The Sword Of Destiny\\""
      videoId="1ke3zayKKm8"
    />
    <li
      className="load-more"
    >
      <Loader />
    </li>
  </ul>
</div>
`;
